### PR TITLE
[PAY-1792] Add formik, zod, and proper inputs to USDC withdrawal modal

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

Add formik, zod, and proper inputs to USDC withdrawal modal

* Add error states to inputs (see screenshot)
* Break upload flow input helpers into a separate util
* Add `setData` method to `createModal` so the modal pages can use the shared modal state to manage internal pages

### How Has This Been Tested?

Local vs. staging, all the way up to the submit

### Screenshots

image

### CODE REVIEW

File: packages/web/src/utils/tokenInput.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  const { whole, decimal, dot } = match?.groups || {}
+
+  // Conditionally render the decimal part, and only for the number of decimals specified
+  const stringAmount = dot
+    ? `${whole}.${(decimal ?? '').substring(0, PRECISION)}`
+    : whole
+  return { human: stringAmount, value: Number(stringAmount) * 100 }
+}
+
+export const onTokenInputBlur = (e: FocusEvent<HTMLInputElement>) => {
```
Comment (dharit-tan) [inline packages/web/src/utils/tokenInput.ts:18]: general comment at top of these fns to explain what they do? what i understand is:
`onTokenInputChange` prevents multiple `.`
`onTokenInputBlur` pads out to 2 decimals.

File: packages/web/src/components/withdraw-usdc-modal/components/TransferSuccessful.tsx [hunk 2]
```diff
@@ -57,11 +65,11 @@ export const TransferSuccessful = () => {
       <div className={styles.destination}>
         <TextRow left={messages.destinationAddress} />
         <Text variant='body' size='medium' strength='default'>
-          {wallet}
+          {addressValue}
         </Text>
         <HarmonyPlainButton
           style={{ padding: 0 }}
-          onClick={() => openExplorer(signature)}
+          onClick={() => openExplorer(signature ?? '')}
```
Comment (dharit-tan) [inline packages/web/src/components/withdraw-usdc-modal/components/TransferSuccessful.tsx:72]: should we just hide this button if the sig doesn't exist for whatever reason?

File: packages/web/src/components/withdraw-usdc-modal/WithdrawUSDCModal.tsx [hunk 3]
```diff
@@ -12,12 +21,53 @@ import { TransferInProgress } from './components/TransferInProgress'
...
+    pleaseConfirm:
+      'Please confirm you have reviewed the details and accept responsibility for any errors resulting in lost funds.'
+  }
+}
+
+export const AMOUNT = 'amount'
+export const ADDRESS = 'address'
+export const CONFIRM = 'confirm'
+
+const WithdrawUSDCFormSchema = (userBalance: number) => {
```
Comment (dharit-tan) [inline packages/web/src/components/withdraw-usdc-modal/WithdrawUSDCModal.tsx:38]: whoa cool. can probably delete the slice/saga fns that validate the input now.

**Review by dharit-tan — APPROVED** [review]

File: packages/web/src/components/withdraw-usdc-modal/components/TransferSuccessful.tsx (hunk 2 above)
Comment (raymondjacobson) [reply packages/web/src/components/withdraw-usdc-modal/components/TransferSuccessful.tsx:72]: I think we probably need an additional error state here. Will confer w/ design. Let's see how this comes together when we wire up the signature bits to the sagas.

File: packages/web/src/components/withdraw-usdc-modal/WithdrawUSDCModal.tsx (hunk 3 above)
Comment (raymondjacobson) [reply packages/web/src/components/withdraw-usdc-modal/WithdrawUSDCModal.tsx:38]: doesn't hurt to do in both places, but i think this is the canonical way of doing input validation now! (learned this yday)

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rj-pay-1792

**Pull request merged**